### PR TITLE
Integrated signout and IsAMatch. Added redux state clean-ups

### PR DESCRIPTION
**Pull request by AleksTeresh**

### CODE REVIEW

File: src/card-queue/components/CardQueue.js [hunk 1]
```diff
@@ -70,38 +74,69 @@ export class CardQueue extends React.PureComponent {
...
+              tripEnd={currCard.endTime}
+              location={{ lat: currCard.lat, lon: currCard.lon }}
+              images={currCard.photos}
+              cardText={currCard.description}
+              locationName={locationName}
+            />
+            <CardButtons style={style.buttons} onTouchTap={this.onButtonTap} />
+            <ItsAMatchOverlay
+              open={match}
+              onContinue={() => actions.currentCard.nextTarget()}
```
Comment (AleksTeresh) [inline src/card-queue/components/CardQueue.js]: right

File: src/card-queue/components/CardQueue.js [hunk 2]
```diff
@@ -114,18 +149,25 @@ export class CardQueue extends React.PureComponent {
...
+    const {
+      queue,
+      current
+    } = this.props
     return (
       current < queue.size
       ? this.makeCard()
-      : (<EmptyCard callback={e => this.props.emptyCallback(e)} />)
+      : (
+        <EmptyCard callback={e => this.props.emptyCallback(e)} />
```
Comment (AleksTeresh) [inline src/card-queue/components/CardQueue.js]: I did probably, but then removed it. That's the reason for the change

File: src/card-queue/components/CardQueue.js [hunk 3]
```diff
@@ -114,18 +149,25 @@ export class CardQueue extends React.PureComponent {
...
+      )
     )
   }
 }
 
 function mapStateToProps (state: AppState) {
   return {
+    profilePhoto: state.profile.profile.photos.first(),
```
Comment (AleksTeresh) [inline src/card-queue/components/CardQueue.js]: yes. 1st profile = profileRelated reducers, 2nd - the actual user profile info itsefl

File: src/card-queue/components/ItsAMatchOverlay.js [hunk 4]
```diff
@@ -59,23 +62,65 @@ const style = {
...
+              : oidToUrl(avatarImgOwn)
+            } />
+          <Avatar
+            className='animated ease slideInRight'
+            style={style.avatar}
+            size={128}
+            src={
+              avatarImgTarget === undefined
+              ? stockAvatar
+              : oidToUrl(avatarImgTarget)
```
Comment (AleksTeresh) [inline src/card-queue/components/ItsAMatchOverlay.js]: It is a proper name indeed. All the images are stored as numbers, which is oid. The function converts an image's oid to an actual url

File: src/card-queue/components/ItsAMatchOverlay.js [hunk 5]
```diff
@@ -86,10 +131,17 @@ class ItsAMatchOverlay extends Component {
             backgroundColor={palette.primary1Color}
             buttonStyle={style.innerButton}
             label={<span style={style.buttonText}>Send a message</span>}
+            onClick={onStartMessaging}
```
Comment (AleksTeresh) [inline src/card-queue/components/ItsAMatchOverlay.js]: True, will you fix it later or should I?

File: src/card-queue/components/PersonalCard.js [hunk 6]
```diff
@@ -16,6 +16,8 @@ import Calendar from 'material-ui/svg-icons/action/date-range'
 import Vehicle from 'material-ui/svg-icons/maps/directions-car'
 import People from 'material-ui/svg-icons/social/people'
 
+import { oidToUrl } from '../../core/photo-utils'
```
Comment (AleksTeresh) [inline src/card-queue/components/PersonalCard.js]: already replied

File: src/card-queue/components/ItsAMatchOverlay.js (hunk 5 above)
Comment (StormBlade97) [inline src/card-queue/components/ItsAMatchOverlay.js]: i will fix it

**Pull request merged**